internal/datastore/mongo_store.go

```diff
@@ -0,0 +1,92 @@
+package datastore
```

Comment:
datastore and persistence packages are redundant. Let's move `internal/persistence/persistence.go internal/datastore/store.go`